[Zhang Bokun] iP

bokun2 opened the pull request:
# DukeProject
A Bullet List for Tasks

- Deadline
- Event
- Todo

A Numbered List for Command

1. ExitCommand
2. DeleteCommand
3. AddCommand
4. OtherCommand
5. etc.

Test Code Block
```ruby
public class TaskTest {
    @Test
    public void testMark() {
        Task t = new Task("todo borrow book");
        assertEquals(" ",t.getStatusIcon());
        t.taskDone();
        assertEquals("X",t.getStatusIcon());
        t.taskUndone();
        assertEquals(" ",t.getStatusIcon());
    }
}
```

TaskList

- [X] Add GUI feature

- [ ] Add varargs

Please leave comments for this PR! 👍 :smiley:

> Ming due to the circumstances, the knower system according to the matter.    -Huan Kuan

The road to [my repo](https://github.com/bokun2/ip).

Split are used in code like `s0.split(" ",2)[1];`

***My design concept about dealing exception caused by inputs***

**To handling exception: using DukeException**
_This will make sure every input has a reasonable response._
~~But seems that this make my input format a mess~~

Commit:
Add Gradle support

Commit:
no message

Commit:
add a new class TaskList

Commit:
add a new class task

Commit:
add 3 new classes( rep 3 new kind of tasks)

Commit:
adding test

Commit:
no message

Commit:
no message

Commit:
Add a FileContainer to save tasks

Commit:
Add time recognization feature

Commit:
Merge branch-Level-7 into master branch

Delete the data directory and the duke.txt as a person new to Duke can use it on his/her own computer.

Commit:
Merge branch-Level-8 with master branch

Fix the merge and conflicts. Fix the content of text-ui-test. However the test can only run once for the duke.txt will produce more answers than before.

Commit:
Use More OOP

Commit:
Organize all classes into packages

Commit:
Add Gradle to the project

Commit:
Add JUnit tests. Correct mistakes in build.gradle in previous commit

Commit:
Add Java Doc to all classes except test classes

Commit:
Modify the code to follow the coding standard

Commit:
Add a way to find a task by searching for a keyword

Commit:
Merge branch 'branch-A-JavaDoc' with master branch

Commit:
Merge branch 'branch-A-CodingStandard' with master branch and solve conflicts

Commit:
Merge branch 'branch-Level-9' with master branch and solve conflicts

Commit:
Check style violation and correct some of them

maxng17 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,33 @@
+     */
+    @Override
+    public void execute(TaskList taskList, Storage storage) throws DukeException {
+        taskList.delete(Integer.valueOf(this.fullCommand.split(" ")[1]),
```
Quite complicated to look at. Might consider splitting it into several lines and using variables to achieve this.

maxng17 commented on src/main/java/duke/command/MarkingCommand.java:
```diff
@@ -0,0 +1,34 @@
+    @Override
+    public void execute(TaskList taskList, Storage storage) throws DukeException {
+        if (this.fullCommand.split(" ")[0].equals("mark")) {
+            taskList.markAsDone(Integer.valueOf(this.fullCommand.split(" ")[1]) - 1, storage);
```
Same as above maybe better to split up the steps instead of squeezing them into 1 line. Hard to read. Applies to the line, 2 lines down too.

maxng17 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,43 @@
+     * @return A certain kind of command waiting for execution.
+     * @throws DukeException Throws DukeException with remind message when the input format is wrong.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```
Quite inefficient to keep splitting the string until you get the correct command. Can consider just doing it once storing it as a variable then checking to see if the word is a command.

maxng17 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,98 @@
+            FileWriter fw = new FileWriter(this.file);
+            for (int i = 0; i < t.size(); i++) {
+                if (t.get(i) != null) {
+                    if (t.get(i).getStatusIcon().equals("X")) {
```
Following code quality guidelines, they don't really like many levels of indentation. Maybe can consider splitting up the if statements so won't have to nest this if statement. Something like:
```
Task task = t.get(i);
if (task == null) {
    continue;
}
if (task.getStatusIcon().equals("X")) {
  ...
} else {
...
}
```

maxng17 commented on src/main/java/duke/task/DeadLine.java:
```diff
@@ -0,0 +1,51 @@
+        } catch (IndexOutOfBoundsException ie) {
+            throw (new DukeException("OOPS!!! The description of a deadline cannot be empty."));
+        } try {
+            super.correctDescrition(description.split("by")[0].split(" ", 2)[1]);
```
Abit long and complicated to read. Might be better breaking up into separate lines.

maxng17 commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,127 @@
+            String res="";
+            for (int i = 0; i < this.taskList.size(); i++) {
+                if (this.taskList.get(i).getDay() != null) {
+                    if (this.taskList.get(i).getDay().equals(d)) {
```
It might be better to have lesser levels of indentation so that the code is easier to read. Can consider splitting up the if statements and not nest them

maxng17 reviewed (commented):
Overall the code is good and achieves the functionalities needed. Due to the strictness of the code quality in the textbooks, had to point out parts which appear hard to read. I think its preferable to use variables and not squeeze 3 functions into the same line. 👍

chao890 commented on src/main/java/duke/task/DeadLine.java:
```diff
@@ -0,0 +1,51 @@
+    public DeadLine(String description) throws DukeException {
+        super("tempTask");
+        try {
+            String temp=description.split(" ")[1];
```
Perhaps you could add spacing before and after your `=` operation?

chao890 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,54 @@
+ * Represents the task of event format.
+ */
+public class Event extends Task {
+    private String at;
```
Could this naming variable be more descriptive? 
```suggestion
    private String dateTime;
```

chao890 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,99 @@
+    private LocalDate day;
+    private String fullDescription;
+    public Task (String description) {
+        this.description=description;
+        this.isDone=false;
```
perhaps you could leave a space before and after your `=` operator?

chao890 commented on src/test/java/duke/task/EventTest.java:
```diff
@@ -0,0 +1,20 @@
+import org.junit.jupiter.api.Test;
+
+import duke.dukeexception.DukeException;
+public class EventTest {
```
Perhaps you could leave an empty line after your last import statement?

chao890 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,99 @@
+     * @return A string of right format describe the task.
+     */
+    public String printTask(){
+        return ("["+this.getStatusIcon()+"]"+" "+this.description);
```
perhaps you could leave a space before and after your `+` operators?

chao890 reviewed (commented):
LGTM! Maybe could work on your spacing for operators, but other than that I like how you separated your classes.

Commit:
Add Ui features. Will be modified later

Commit:
Create a Jar file

Commit:
Add a new branch for level 10 (which is implemented in master branch)to satisfy the dashboard

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add a new branch for level 10 (which is implemented in master branch)to satisfy the dashboard

Commit:
Create a new fat jar

Commit:
Add assertions

bokun2 committed:
Merge pull request #2 from bokun2/branch-A-Assertions

A-Assertions

Commit:
Improve code quality

Some methods are too long and hard to read.
The split methods make the code hard to read.

Let's
* shorten the method
* Use variable for string indication

More code quality be improved in future commits

bokun2 committed:
Merge pull request #3 from bokun2/branch-A-CodeQuality

Improve code quality

Commit:
Merge branch 'master' of https://github.com/bokun2/ip into branch-A-CodeQuality

* 'master' of https://github.com/bokun2/ip:

Commit:
Set up CI

Add .github/workflows/gradle.yml file.

Github actions will run the workflow during project events (PR updating, branch updating).

Commit:
Revert "Create a new fat jar"

This reverts commit 3cd6ac14e37439e7d8025526ed1749fce986218a.

Commit:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Set up CI
  Improve code quality
  Add assertions

Commit:
Add an extension

Improve the performance of searching.

When typing "find" or equivalent, Duke can find items even if the keyword matches the item only partially.

Duke will return task containing the keywords.

Support to find certain type of task(e.g. event).

Commit:
Merge branch 'branch-C-BetterSearch'

* branch-C-BetterSearch:
  Add an extension

Commit:
Add a better GUI

Change profile picture to circle. Change dialog box to a more readable pattern.

Respond in grey when no exception and in red if the input is wrong.

Commit:
Add more exceptions handlings

Handle the case when loading task is wrong.

Commit:
Add showcase

Fix bugs and add showcase picture to Ui.png in docs

Commit:
Add more Junit tests

Add tests for ToDo class and add more tests for other commission.

Commit:
Finalize the feature

Add more java doc and correct code according to code style.

Commit:
Add a user guide

Commit:
Add a screenshot

Commit:
Correct bugs in user guide

Commit:
Change some user guide features

Commit:
Create a jar

Commit:
Convert img to jpg

Try to reload the img for it is not shown normally on linux and fix bugs in user guide.

Commit:
Fix some bugs

Commit:
Fix some bugs

Commit:
Update user guide
